Engine/ScriptAnalyzer.cs

```diff
@@ -490,7 +489,10 @@ internal bool ParseProfile(object profileObject, PathIntrinsics path, IOutputWri
             {
                 return hasError;
             }
-            foreach (var settingKey in settings.Keys)
+
+            var settingsKeys = new string[settings.Keys.Count];
+            settings.Keys.CopyTo(settingsKeys, 0);
```

Comment:
Assigning elements to the dictionary while iterating over it was causing problems. After creating a separate copy of the keys the problem went way.